lib/bolt/inventory.rb

```diff
@@ -345,13 +346,18 @@ def set_plugin_hooks(tname, hash)
     end
     private :set_plugin_hooks
 
-    def remove_node(current_group, target, desired_group)
+    def remove_node(current_group, target, desired_group, track = { 'all' => nil })
+      # Remove the target from the group
       if current_group.name == desired_group
         current_group.nodes.delete(target.name)
+      # If the target remains in the group, add the group data to the cache
+      else
+        add_group_data_to_cache(current_group, target, track)
```

Comment:
We should only do this if the target is actually in the group. Otherwise this winds up adding the config from _every_ group to the target.